module/LineTracer.cpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file LineTracer.cpp
+ * @brief ライントレースする
+ * @author yutotanaka24,kodama0720
+ */
+#include "LineTracer.h"
+
+LineTracer::LineTracer(bool _isLeft) : edge((_isLeft == true) ? 1 : -1) {}
+
+//設定された距離を走行する
+void LineTracer::run(double targetDistance, int targetBrightness, int pwm, PidGain& gain)
+{
+  double currentDistance = 0;
+  double currentPid = 0;
+  Pid pid(gain.kp, gain.ki, gain.kd, targetBrightness);
+
+  //走行距離のリセット
+  controller.reset();
+
+  //走行距離が目標距離を超えるまで繰り返す
+  while(true) {
+    currentDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
+    if(currentDistance > targetDistance) {
+      break;
+    }
+    // PID計算
+    currentPid = pid.calculatePid(measurer.getBrightness()) * edge;
+    //モータのPWM値をセット
+    controller.setRightMotorPwm(pwm - currentPid);
+    controller.setLeftMotorPwm(pwm + currentPid);
+    // 10ミリ秒待機
+    controller.sleep();
+  }
+}
```

Comment:
controller.reset();で走行距離がリセットされていないようなのですが、原因がわかりません。一応、去年のソースコードを参考に以下のように書き換えたら動くことは確認できましたが、できればAPIの関数を使いたいです。どうすればよいでしょうか。
```suggestion
void LineTracer::run(double targetDistance, int targetBrightness, int pwm, const PidGain& gain)
{
  int initialDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
  int currentDistance = initialDistance;  // 現在の走行距離
  double currentPid = 0;
  Pid pid(gain.kp, gain.ki, gain.kd, targetBrightness);

  //走行距離が目標距離を超えるまで繰り返す
  while(true) {
    currentDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
    if(currentDistance - initialDistance > targetDistance) {
      break;
    }
    // PID計算
    currentPid = pid.calculatePid(measurer.getBrightness()) * edge;
    //モータのPWM値をセット
    controller.setRightMotorPwm(pwm - currentPid);
    controller.setLeftMotorPwm(pwm + currentPid);
    currentDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
    // 10ミリ秒待機
    controller.sleep();
  }
}
```

Reply:
reset前後で、getCountで返ってくる値が変わらないってこと？

Reply:
多分だけど，Controllerクラスのreset関数が問題なんじゃないかな
去年（と確か一昨年も）同じような問題があったんだけど，タスク周期の問題？でリセットしきれないから，下のコードみたいにして両輪のカウントが0になるまでwhile文回してたんだよね
これでできないか確認してみて
```
void Controller::resetMotorCount(){
  while((this->getLeftMotorCount() != 0) || (this->getRightMotorCount() != 0)) {
    leftWheel.reset();
    rightWheel.reset();
    this->tslpTsk(20000);
  }
}
```

Reply:
動画では、どのコードで走らせているの？
プルリクで出されているコード？それとも、児玉君が書いてるコード？

Reply:
プルリクのコードでrun関数の呼び出し1回で走らせています。